### DISCUSSION TITLE
How do I hide a slide using pptx export?

### DISCUSSION THREAD

**Discussion by jwhendy** [quarto-dev/quarto-cli#9684] [Q&A] [answered]
### Description

Hello, 

This seemed simple enough, but I'm striking out. I want to author a pptx containing python code, and simply wanted to not export some of the setup code. Things I can find:

- how to have [a hidden div](https://quarto.org/docs/authoring/markdown-basics.html#equations)
- didn't notice it in the docs, but [via SO](https://stackoverflow.com/questions/76795848/quarto-how-to-hide-headings-in-html-render-but-keep-in-rstudio-document) this hidden approach also works for html sections
- how to [set visibility](https://quarto.org/docs/presentations/revealjs/advanced.html#slide-visibility) in a revealjs presentation

`````
---
title: "Presentation"
format:
  pptx:
    reference-doc: template.pptx
---

::: {.hidden}

```python
print('this output and whatever div/section/slide contains it will not be included')
```
:::

# setup code {.hidden}

```python
print('this output and whatever div/section/slide contains it will not be included')
```

# Slide with stuff

- some stuff
- some other stuff
- this stuff will be included on a slide

`````

Desired output of this example would be a title slide and the slide with stuff. I included mocked up approaches I tried with the code slides, but so far they result in a slide containing their content. I'm sure it's something silly I'm missing, but after ~30min of tinkering figured I'd be better off asking for help!

**Comment by mcanouil** [comment, marked as answer]
Edit: 
- `hidden` is not for "don't include the content" (also I believe this to be for LaTeX and HTML formats only).
- "slide visibility" is a Reveal.js feature which has no equivalent in Powerpoint

You are left with conditional content: https://quarto.org/docs/authoring/conditional.html

**Reply by jwhendy** [reply]
Thanks. From my 2/3 links to the docs, hopefully it's clear that I did look at the documentation :)

For what it's worth as a new user, part of the need to ask this question is related to the different terms used to describe it, and how to determine what's output specific vs. global to all formats. After trying some other ways to hide, I suspected I needed something pptx specific.

I appreciate the pointer, and confirm that using `{.content-hidden}` does indeed do the trick.

**Reply by mcanouil** [reply]
Indeed, it can be tricky to find the right tool/option when each formats have their own specificity.
Quarto tries to standardise as much as possible with features such as the conditional content.

**Reply by jwhendy** [reply]
Now that I know where this functionality lies, I think "conditional content" is a non-obvious description, as one can use it to hide content, period. The doc page never shows `{.content-hidden}` on its own, only with a trailing format specifier, even though this works fine.

It isn't clear to me why `::: {.hidden} :::` (mentioned out of the blue without context for LaTeX macros) is LaTeX specific while `::: {.content-hidden} :::` is global? Not that anyone asked, just thinking out loud as a noob encountering the docs with fresh eyes :)

I could see the conditional content page being generalized to "Hiding/showing content" to explain the different ways one can do this (and which ways work with what formats, perhaps in a table), and then include the sub-case in which one might want to hide/show content only for certain formats.

If any of these things are appealing/seem reasonable, I would be game to draft a PR.

**Reply by cscheid** [reply]
I agree that we should be doing a better job of describing these, but here goes the difference:

`.hidden` means "put it on the resulting page, but make it invisible" (useful in the case you linked to, because otherwise MathJax can't see those statements).

`.content-hidden` means "remove it from the document entirely".

We're unfortunately stuck with these names until we're willing to do a change that breaks backwards compatibility.
